packages/cli/src/lib/project/project.ts

```diff
@@ -0,0 +1,57 @@
+import {createWriteStream, existsSync, unlinkSync} from 'fs';
+import {join} from 'path';
+import {cli} from 'cli-ux';
+import * as archiver from 'archiver';
+
+export class Project {
+  constructor(private pathToProject: string) {
+    this.ensureProjectCompliance();
+  }
+
+  private get pathToTemporaryZip() {
+    return join(this.pathToProject, 'snapshot.zip');
+  }
+
+  private get pathToResources() {
+    return join(this.pathToProject, 'resources');
+  }
+
+  deleteTemporaryZipFile() {
+    unlinkSync(this.pathToTemporaryZip);
+  }
+
+  ensureProjectCompliance() {
+    /*
+     * TODO: CDX-354: add checks to ensure the project is indeed a valid project
+     * e.g. * Check if path to resources is a folder
+     *      * Check if the root has a valid config file
```

Comment:
Possibly means we will need a `repair` command ? That's not 100% mandatory by the way... 

As end user, you could always get something valid from your git history anyway.